### PR TITLE
Feature/messaging

### PR DESCRIPTION
**Pull request by jaba9531**

### CODE REVIEW

File: client/src/components/messagePage/index.jsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    this.socket = io('http://localhost:5000');
+  }
+
+  componentWillMount() {
+    let channels = this.props.channels;
+    let username = this.props.user.username;
+    let messages = this.props.channels.messages;
+    let tempArr = [];
+    for (var i = 0; i < channels.length; i++) {
+      var str = channels[i].participants.replace(username, "").replace(":", "");
```
Comment (jaba9531) [inline client/src/components/messagePage/index.jsx]: note to self get rid of var

File: client/src/components/messagePage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        this.setState({toggleModal: false, channel: channel})
+      } else {
+        this.setState({toggleModal: false});
+      }
+    } else {
+      this.setState({toggleModal: true});
+    }
+  }
+
+  handleChannelButtonClick() {
```
Comment (zgallagher08) [inline client/src/components/messagePage/index.jsx]: rename handleAddChatButton

File: client/src/components/messagePage/index.jsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import * as colors from 'material-ui/styles/colors';
+import ContentAdd from 'material-ui/svg-icons/content/add';
+import MessagePageModal from './messagePageModal.jsx';
+import io from 'socket.io-client';
+
+class MessagePage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      toggleModal: false,
```
Comment (zgallagher08) [inline client/src/components/messagePage/index.jsx]: rename state toggleModal to toggleAddChatModal (and all references)

File: client/src/components/messagePage/index.jsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    } else {
+      this.setState({toggleModal: true});
+    }
+  }
+
+  handleChannelButtonClick() {
+    this.setState({toggleModal: true});
+  }
+
+  handleChannelNameClick(value) {
```
Comment (zgallagher08) [inline client/src/components/messagePage/index.jsx]: rename parameter value to channelName

File: client/src/components/messagePage/index.jsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+      }
+    }
+    this.setState({activeChannel: value, messages: tempMessagesArr});
+  }
+
+  handleMessageInputValueChange(e) {
+    this.setState({messageInput: e.target.value});
+  }
+
+  handleSubmitMessageButtonClick() {
```
Comment (zgallagher08) [inline client/src/components/messagePage/index.jsx]: rename handleSubmitMessage

File: client/src/components/messagePage/index.jsx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+      <div className="messageView">
+        <div id="left">
+          <div id="left-header-container">
+            <span id="left-header">
+              Messages
+            </span>
+            <FloatingActionButton onClick={this.handleChannelButtonClick} className="add-chat" backgroundColor={colors.grey600} mini={true}><ContentAdd /></FloatingActionButton>
+            <div className="channelScrollBox">
+            {
+              this.state.channelNameList.map(value =>
```
Comment (zgallagher08) [inline client/src/components/messagePage/index.jsx]: add a key using channelName

File: client/src/components/messagePage/index.jsx [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+            :
+            <div>
+              No Channel Currently Selected
+            </div>
+          }
+          </div>
+        </div>
+        <div id="right-middle">
+          {
+            this.state.messages.map(value =>
```
Comment (zgallagher08) [inline client/src/components/messagePage/index.jsx:141]: add a key using shortid.generate()

File: client/src/components/messagePage/index.jsx [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+                  {value.text}
+                </span>
+              </div>
+            )
+          }
+        </div>
+        <div id="right-bottom">
+        <div>
+          <label>
+            <input id="messageInput" type="text" value={this.state.messageInput} onChange={this.handleMessageInputValueChange} onKeyPress={(e) => this.handleKeyPress(e)}/>
```
Comment (zgallagher08) [inline client/src/components/messagePage/index.jsx]: try to consolidate to using just onKeyPress and test if e.key is 'Enter'

rename function to this.handleMessageInput

File: client/src/components/messagePage/messagePageModal.jsx [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+  async handleSubmitButtonClick() {
+    let newUser = this.state.inputValue;
+    let activeUser = this.props.activeUser;
+    let tempArr = [newUser, activeUser];
+    let newArr = tempArr.sort();
+    let channelStr = tempArr[0] + ':' + tempArr[1];
+    let payload = {channelStr: channelStr, newUser: newUser}
+    try {
+      const result = await axios.post('http://localhost:8000/api/messages/getchannels/addchannel', payload)
```
Comment (zgallagher08) [inline client/src/components/messagePage/messagePageModal.jsx]: add cookie to axios post

**Review by zgallagher08 — CHANGES_REQUESTED** [review]
looks good, small changes

**Review by amelendez37 — DISMISSED** [review]

**Review by zgallagher08 — APPROVED** [review]

**Pull request merged**